### PR TITLE
Fix for gateway alias resolution errors

### PR DESCRIPTION
**Pull request by jonnydgreen**
Fixes: https://github.com/mercurius-js/mercurius/issues/428

I have added tests for both gateway and non-gateway modes of operation. Next, I'll run the benchmarks and update here to make sure that this is the right approach.

### CODE REVIEW

File: test/aliases.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  const query = `
+    query {
+      user: me {
+        id
+        name
+        newName: name
+        otherName: name
+        quote(input: "quote")
+        firstQuote: quote(input: "foo")
+        secondQuote: quote(input: "bar")
```
Comment (mcollina) [inline test/aliases.js:72]: I didn't even know this syntax was possible, I've learned something new today!

**Review by mcollina — COMMENTED** [review]
Looks good to me!

File: test/aliases.js (hunk 1 above)
Comment (jonnydgreen) [reply test/aliases.js:72]: Awesome!! :) Yeah I've found it to be super useful when you want multiple results from the same field in a single request!

Comment (PabloSzx) [reply test/aliases.js:72]: a little bit of shameless plug but in [gqless](https://github.com/gqless/gqless) we use them to keep track of differences in variables in fields, they are very useful

Comment (jonnydgreen) [reply test/aliases.js:72]: That's really neat!

**Comment by jonnydgreen** [comment]
Just ran some benchmarks and all looks good I think!

### Master
```
Running 60s test @ http://localhost:4000/graphql
100 connections
50 workers

┌─────────┬───────┬───────┬───────┬───────┬──────────┬─────────┬────────┐
│ Stat    │ 2.5%  │ 50%   │ 97.5% │ 99%   │ Avg      │ Stdev   │ Max    │
├─────────┼───────┼───────┼───────┼───────┼──────────┼─────────┼────────┤
│ Latency │ 14 ms │ 20 ms │ 30 ms │ 36 ms │ 20.27 ms │ 6.91 ms │ 640 ms │
└─────────┴───────┴───────┴───────┴───────┴──────────┴─────────┴────────┘
┌───────────┬────────┬────────┬────────┬─────────┬─────────┬────────┬────────┐
│ Stat      │ 1%     │ 2.5%   │ 50%    │ 97.5%   │ Avg     │ Stdev  │ Min    │
├───────────┼────────┼────────┼────────┼─────────┼─────────┼────────┼────────┤
│ Req/Sec   │ 1772   │ 3155   │ 4807   │ 5491    │ 4815.14 │ 590.57 │ 1772   │
├───────────┼────────┼────────┼────────┼─────────┼─────────┼────────┼────────┤
│ Bytes/Sec │ 362 kB │ 644 kB │ 980 kB │ 1.12 MB │ 982 kB  │ 120 kB │ 361 kB │
└───────────┴────────┴────────┴────────┴─────────┴─────────┴────────┴────────┘
```

### This PR

```
Running 60s test @ http://localhost:4000/graphql
100 connections
50 workers

┌─────────┬───────┬───────┬───────┬───────┬──────────┬─────────┬────────┐
│ Stat    │ 2.5%  │ 50%   │ 97.5% │ 99%   │ Avg      │ Stdev   │ Max    │
├─────────┼───────┼───────┼───────┼───────┼──────────┼─────────┼────────┤
│ Latency │ 14 ms │ 20 ms │ 30 ms │ 36 ms │ 20.18 ms │ 6.66 ms │ 592 ms │
└─────────┴───────┴───────┴───────┴───────┴──────────┴─────────┴────────┘
┌───────────┬────────┬────────┬────────┬─────────┬─────────┬────────┬────────┐
│ Stat      │ 1%     │ 2.5%   │ 50%    │ 97.5%   │ Avg     │ Stdev  │ Min    │
├───────────┼────────┼────────┼────────┼─────────┼─────────┼────────┼────────┤
│ Req/Sec   │ 1747   │ 3235   │ 4875   │ 5547    │ 4834.24 │ 596.51 │ 1747   │
├───────────┼────────┼────────┼────────┼─────────┼─────────┼────────┼────────┤
│ Bytes/Sec │ 357 kB │ 660 kB │ 994 kB │ 1.13 MB │ 986 kB  │ 122 kB │ 356 kB │
└───────────┴────────┴────────┴────────┴─────────┴─────────┴────────┴────────┘
```

**Review by PabloSzx — APPROVED** [review]
👍

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**